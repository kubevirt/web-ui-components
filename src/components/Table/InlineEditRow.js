import React from 'react';
import { Table } from 'patternfly-react';

// HACK: DragableRow gives this error:
//   Stateless function components cannot be given refs. Attempts to access this ref will fail.
// eslint-disable-next-line react/prefer-stateless-function
export class InlineEditRow extends React.Component {
  render() {
    return <Table.InlineEditRow {...this.props} />;
  }
}
