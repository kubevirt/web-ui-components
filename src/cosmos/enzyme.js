import createContext from 'react-cosmos-test/enzyme';

import paths from '../../config/paths';

export default function createTestContext(args) {
  args.proxies = args.proxies || []; // workaround of react-cosmos-test bug
  return createContext({
    ...args,
    cosmosConfigPath: `${paths.config}/cosmos.config.js`,
  });
}
