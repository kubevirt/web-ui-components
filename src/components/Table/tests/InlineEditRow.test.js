import React from 'react';
import { shallow } from 'enzyme';
import { Table } from 'patternfly-react';

import { InlineEditRow } from '../InlineEditRow';

const testProp = 'testProp';

const testInlineEditRow = () => <InlineEditRow testProp={testProp} />;

describe('<InlineEditRow />', () => {
  it('renders correctly', () => {
    const component = shallow(testInlineEditRow());
    expect(component).toMatchSnapshot();
  });

  it('passes props', () => {
    const component = shallow(testInlineEditRow());
    expect(component.find(Table.InlineEditRow).props().testProp).toBe(testProp);
  });
});
