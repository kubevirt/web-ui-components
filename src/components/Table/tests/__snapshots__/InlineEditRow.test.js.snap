// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InlineEditRow /> renders correctly 1`] = `
<TableInlineEditRow
  buttonsClassName=""
  buttonsPosition={[Function]}
  isEditing={[Function]}
  last={false}
  messages={
    Object {
      "cancelButtonLabel": "Cancel",
      "confirmButtonLabel": "Save",
    }
  }
  onCancel={[Function]}
  onConfirm={[Function]}
  testProp="testProp"
/>
`;
